python/rmm/_lib/device_buffer.pyx

```diff
@@ -278,18 +278,25 @@ cdef class DeviceBuffer:
 
         return b
 
-    cdef size_t c_size(self):
+    cdef size_t c_size(self) except *:
         return self.c_obj.get()[0].size()
 
-    cpdef void resize(self, size_t new_size):
+    cpdef void resize(self, size_t new_size) except *:
         self.c_obj.get()[0].resize(new_size)
 
-    cpdef size_t capacity(self):
+    cpdef size_t capacity(self) except *:
         return self.c_obj.get()[0].capacity()
 
-    cdef void* c_data(self):
+    cdef void* c_data(self) except *:
         return self.c_obj.get()[0].data()
 
+    cdef device_buffer c_release(self) except *:
```

Comment:
I think this should be `except +` if we are trying to wrap C++ exceptions. `except *` is for checking whether C/C++ code (not Cython) raises an exception (like when using Python's C API directly).

```suggestion
    cdef device_buffer c_release(self) except +:
```

Reply:
Actually scratch that. The [`.release()` method is `noexcept`]( https://en.cppreference.com/w/cpp/memory/unique_ptr/release ). So this shouldn't be needed either.

Reply:
In `c_release()` we are also doing `self.ptr = 0`, which threw for me initially as `ptr` was a `property`. That exception got swallowed. In general, I think it's safest to `except *` anywhere that we are returning a non-Python object

Reply:
> In general, I think it's safest to except * anywhere that we are returning a non-Python object

Ah right. Not sure why I missed this. Yeah agreed.